[924] Revert withdrawals (support)

avinhurry opened the pull request:
### Context

We often see providers unintentionally withdrawing courses, so we are building the ability to revert withdrawn courses as a feature into our support UI.

### Changes proposed in this pull request

- Show course status on all courses within support

- Show a `Revert withdrawal` link below the course status if the course status is `Withdrawn`

- When the course is reverted, the `course_status` is set to `published` and the `last_published_timestamp_utc` is set to the current time.

### Screenshots

<img width="1081" alt="image" src="https://github.com/DFE-Digital/publish-teacher-training/assets/50492247/4780118b-d601-47d0-9a5d-5e5fa59df5d7">
<img width="1075" alt="image" src="https://github.com/DFE-Digital/publish-teacher-training/assets/50492247/4b78674a-b931-4206-9412-06947e085c9c">
<img width="1157" alt="image" src="https://github.com/DFE-Digital/publish-teacher-training/assets/50492247/255e465f-4091-41ea-9946-e5b4f2377d64">


### Guidance to review

- Go to Publish, withdraw a course.
- Navigate to the same course on support.
- Revert the withdrawal.
- Go back to Publish and ensure the course is back how it was.
- Try to revert the withdrawal of a course that isn't withdrawn (there should be no link).

### Checklist

- [x] Make sure all information from the Trello card is in here
- [x] Attach to Trello card
- [x] Rebased main
- [x] Cleaned commit history
- [x] Tested by running locally
- [x] Inform data insights team due to database changes

avinhurry commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Publish support: Revert withdrawals](https://trello.com/c/luWFpBP2/924-publish-support-revert-withdrawals)

MaeveRoseveare commented on app/views/support/revert_withdrawal/edit.html.erb:
```diff
@@ -0,0 +1,30 @@
+
+ <p class="govuk-body">Reverting this withdrawal will:</p>
+    <ul class="govuk-list govuk-list--bullet">
+      <li>Set the course to published</li>
```
```suggestion
      <li>set the course to published</li>
```

avinhurry replied:
https://github.com/DFE-Digital/publish-teacher-training/pull/3933/commits/fe8cca29f23d5e9ffeca4718e2b7f7fba1577687

MaeveRoseveare commented on app/views/support/revert_withdrawal/edit.html.erb:
```diff
@@ -0,0 +1,30 @@
+ <p class="govuk-body">Reverting this withdrawal will:</p>
+    <ul class="govuk-list govuk-list--bullet">
+      <li>Set the course to published</li>
+      <li>Mark the course as closed for applications</li>
```
```suggestion
      <li>mark the course as closed for applications</li>
```

MaeveRoseveare reviewed (commented):
I just changed the bullet list items to start with lower case. Then also when I reverted the withdrawal, the course status updated to 'Closed' instead of 'Open' is this right? I used the course 'Computing N387' offered by Avanti Schools trust (4A3). 
https://publish-review-3933.test.teacherservices.cloud/support/2024/providers/19862/courses/13001031/edit

the same thing happened for another course I tried: https://publish-review-3933.test.teacherservices.cloud/support/2024/providers/19408/courses/12990146/edit

avinhurry commented:
> I just changed the bullet list items to start with lower case. Then also when I reverted the withdrawal, the course status updated to 'Closed' instead of 'Open' is this right? I used the course 'Computing N387' offered by Avanti Schools trust (4A3). https://publish-review-3933.test.teacherservices.cloud/support/2024/providers/19862/courses/13001031/edit
> 
> the same thing happened for another course I tried: https://publish-review-3933.test.teacherservices.cloud/support/2024/providers/19408/courses/12990146/edit

@MaeveRoseveare , yes it's right, we decided to default to closed when we revert. I'll send you the thread.
